File: app/src/main/java/com/dominikgaller/alpinebooking/shared/outbound/clock/SystemClockPort.java
package com.dominikgaller.alpinebooking.shared.outbound.clock;

import com.dominikgaller.alpinebooking.shared.outport.ClockPort;

import java.time.Instant;

/**
 * Production implementation of {@link ClockPort} that delegates to {@link Instant#now()}.
 *
 * <p>SDD: See {@code documentation/ports/clock.outport.spec.md}.
 */
public class SystemClockPort implements ClockPort {

    @Override
    public Instant now() {
        return Instant.now();
    }
}

File: src/main/kotlin/com/example/service/shared/outbound/clock/SystemClockPort.kt
package com.example.service.shared.outbound.clock

import com.example.service.shared.outport.ClockPort
import java.time.Instant

/**
 * Production implementation of [ClockPort] that delegates to [Instant.now].
 *
 * SDD: see `documentation/ports/clock.outport.spec.md`.
 */
class SystemClockPort : ClockPort {
    override fun now(): Instant = Instant.now()
}
